### PR TITLE
add web3 package and settings

### PR DESCRIPTION
**Pull request by luchopcerra**
## Description
add package for web3 development

## Solution Adopted
based on [this development ](https://github.com/wakeuplabs-io/labitconf-workshop) we are adding web3 package for templating

### CODE REVIEW

File: .github/workflows-off/slither.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    permissions:
+      contents: read
+      security-events: write
+    env:
+      NETWORK_TESTNET: "optimism-sepolia"
+      NETWORK_MAINNET: ${{ secrets.NETWORK_MAINNET }}
+      TESTNET_PRIVATE_KEY: ${{ secrets.TESTNET_PRIVATE_KEY }}
+      MAINNET_PRIVATE_KEY: ${{ secrets.MAINNET_PRIVATE_KEY }}
+      ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
+      ETHERSCAN_BASE_SEPOLIA_API_KEY: ${{ secrets.ETHERSCAN_BASE_SEPOLIA_API_KEY }}
```
Comment (juandahl) [inline .github/workflows-off/slither.yml:22]: should be the same apikey? I think ETHERSCAN_BASE_SEPOLIA_API_KEY is ETHERSCAN_API_KEY

File: packages/web3/contracts/foobar/contract.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @title Foobar
+ * @dev Simple ERC20 Token example, with an initial supply sent to the deployer
+ */
+contract Foobar is ERC20, IErrors, Ownable, ILabitConfToken {
+    uint256 public maxSupply = 1_000_000 * 10 ** decimals();
+
+    /**
+     * @dev Constructor that gives msg.sender all of the initial supply.
+     */
+    constructor() ERC20("LabitConfToken", "LCT24") Ownable(msg.sender) {}
```
Comment (juandahl) [inline packages/web3/contracts/foobar/contract.sol:20]:
```suggestion
    constructor() ERC20("Foo contract", "FOO") Ownable(msg.sender) {}
```

File: packages/web3/shared/helpers/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  try {
+    const { labitConfToken } = await ignition.deploy(tokenModule);
+    if (!labitConfToken) throw new Error("Token deployment failed.");
+
+    return (labitConfToken as DeployResult).address;
+  } catch (error) {
+    console.error("Error deploying token:", error);
+    throw error;
+  }
+};
```
Comment (juandahl) [inline packages/web3/shared/helpers/deploy.ts:40]: maybe simplify this file. Just deploy FooContract

**Review by juandahl — COMMENTED** [review]
LGTM. Just minor changes

File: packages/web3/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+    "name": "@wakeuplabs-io/web3",
+    "version": "1.0.0",
+    "description": "Package for web3 related code like contracts and utils",
+    "main": "index.js",
+    "scripts": {
+      "compile": "npx hardhat compile && npm run abis",
```
Comment (MatiasBarro) [inline packages/web3/package.json:7]: Add `abi` script

File: packages/web3/artifacts/contracts/foobar/contract.sol/Foobar.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (MatiasBarro) [inline packages/web3/artifacts/contracts/foobar/contract.sol/Foobar.json:1]: abi json files should be included in repository?

File: packages/web3/networks.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import envParsed from "./envParsed";
+
+const TESTNET_PRIVATE_KEY = envParsed().TESTNET_PRIVATE_KEY;
+const MAINNET_PRIVATE_KEY = envParsed().MAINNET_PRIVATE_KEY;
+const API_KEY = envParsed().ETHERSCAN_API_KEY;
+
+const sepolia = {
```
Comment (MatiasBarro) [inline packages/web3/networks.ts:7]: Is this network ETH Sepolia? In that case the chainId and urls should be changed

File: packages/web3/artifacts/contracts/foobar/contract.sol/Foobar.json (hunk 5 above)
Comment (luchopcerra) [reply packages/web3/artifacts/contracts/foobar/contract.sol/Foobar.json:1]: its imported for the [wagmi.config.ts](https://github.com/wakeuplabs-io/react-monorepo-template/blob/5b88114b50e1a82fb87fc1834464ff673397f5d5/packages/web3/wagmi.config.ts)

File: packages/web3/networks.ts (hunk 6 above)
Comment (luchopcerra) [reply packages/web3/networks.ts:7]: done!

File: .github/workflows-off/slither.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+name: Slither Analysis
```
Comment (juandahl) [inline .github/workflows-off/slither.yml:1]: I would only add the command. Think that it is not necessary run it  in all commit

**Pull request merged**